File: SolvdQuestions/3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        Set<Character> set = new HashSet<>();

        int maxCount = 0;

        int i=0, j=0;
        int n = s.length();

        while (j < n) {
            char ch = s.charAt(j);

            while (set.contains(ch)) {
                set.remove(s.charAt(i));
                i++;
            }

            set.add(ch);

            j++;
            maxCount = Math.max(maxCount, j-i);
        }

        return maxCount;
    }
}

File: SolvdQuestions/3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.go
func max(a, b int) int {
    if a>b {
        return a
    }
    return b
}

func lengthOfLongestSubstring(s string) int {

    mp := make(map[byte]struct{})

    maxLen, i, j := 0, 0, 0

    for j < len(s) {
        
        if _, exists := mp[s[j]]; !exists {
            mp[s[j]] = struct{}{}
        } else {

            for _, exists := mp[s[j]]; exists; _, exists = mp[s[j]] {
                delete(mp, s[i])
                i++
            }
        }

        mp[s[j]] = struct{}{}

        maxLen = max(maxLen, j-i+1)
        j++
    }
    
    return maxLen
}